docs/docs/cmd/setup.md

```diff
@@ -0,0 +1,81 @@
+# setup
+
+Sets up CLI for Microsoft 365 based on your preferences
+
+## Usage
+
+```sh
+m365 setup [options]
+```
+
+## Options
+
+`--interactive`
+: Configure CLI for Microsoft 365 for interactive use without prompting for additional information
+
+`--scripting`
+: Configure CLI for Microsoft 365 for use in scripts without prompting for additional information
+
+--8<-- "docs/cmd/_global.md"
+
+## Remarks
+
+The `m365 setup` command is an interactive wizard that helps you configure the CLI for Microsoft 365 for your needs. It will ask you a series of questions and based on your answers, it will configure the CLI for Microsoft 365 for you.
+
+The command will ask you the following questions:
+
+- _How do you plan to use the CLI?_
+  You can choose between **interactive** and **scripting** use. In interactive mode, the CLI for Microsoft 365 will prompt you for additional information when needed, automatically open links browser, automatically show help on errors and show spinners. In **scripting** mode, the CLI will not use interactivity to prevent blocking your scripts.
+- _Are you going to use the CLI in PowerShell?_ (asked only when you chose to configure CLI for scripting)
+  To simplify using CLI in PowerShell, you can configure the CLI to output errors to stdout, instead of the default stderr which is tedious to handle in PowerShell.
+- _How experienced are you in using the CLI?_
+  You can choose between **beginner** and **proficient**. If you're just starting working with the CLI, it will show you full help information. For more experienced users, it will only show information about commands' options. No matter how you configure this setting, you can always invoke full help by using `--help full`.
+
+After you answer these questions, the command will build a preset and ask you to confirm using it to configure the CLI.
+
+If you want to configure the CLI on a non-interactive system, or want to configure CLI without answering the questions, you can use the `--interactive` or `--scripting` options. When using these options, CLI will apply the correct presets. Additionally, it will detect if it's used in PowerShell and configure CLI accordingly.
+
+The `m365 setup` command uses the following presets:
+
+- interactive use:
+  - autoOpenLinksInBrowser: true,
```

Comment:
On my end this looks like one single unordered list without indentations. Let's make that more readable.

![image](https://github.com/pnp/cli-microsoft365/assets/5267487/0ce1a4ab-9d5a-45f5-802c-4826f59af580)

Reply:
It turns out that it's a known issue in MkDocs which expects the second level to be indented with 4 spaces, rather than 2. Changing indenting to 4 shows correct rendering in MkDocs but leads to linting errors, plus it doesn't look great in the terminal:

<img width="471" alt="image" src="https://github.com/pnp/cli-microsoft365/assets/11164679/bf05ae79-3cdb-4da6-b193-3d33a5d77b26">

Since we're migrating to Docusaurus, I suggest we leave the indenting as-is and revisit after migration.

Reply:
That's perfectly fine. Will you log an issue on GH @waldekmastykarz?